apps/site/pages/en/learn/modules/package-configuration.mdx

```diff
@@ -0,0 +1,597 @@
+
+- Simple bundler configuration
+
+Cons:
+
+- Larger package weight (basically double)
+- Vulnerable to [the Dual-Package Hazard](#the-dual-package-hazard)
+
+**Working example**: [cjs-with-dual-distro (double)](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/dual/double-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-with-full-dual-distro",
+  "exports": {
+    ".": {
+      "import": "./dist/esm/index.mjs",
+      "require": "./dist/cjs/index.js",
+      "default": "./dist/cjs/index.js"
+    }
+  }
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-with-full-dual-distro",
+  "type": "commonjs",
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": {
+      "import": "./dist/esm/index.mjs",
+      "require": "./dist/cjs/index.js",
+      "default": "./dist/cjs/index.js"
+    },
+    "./package.json": "./package.json"
+  }
+}
+```
+
+Alternatively, you can use `"default"` and `"node"` keys, which are less counter-intuitive: Node.js will always choose the `"node"` option (which always works), and non-Node.js tooling will choose `"default"` when configured to target something other than node. **This precludes the dual-package hazard.**
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-with-alt-full-dual-distro",
+  "exports": {
+    ".": {
+      "node": "./dist/cjs/index.js",
+      "default": "./dist/esm/index.mjs"
+    }
+  }
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-with-alt-full-dual-distro",
+  "type": "commonjs",
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": {
+      "node": "./dist/cjs/index.js",
+      "default": "./dist/esm/index.mjs"
+    },
+    "./package.json": "./package.json"
+  }
+}
+```
+
+### ESM source with only CJS distribution
+
+We're not in Kansas anymore, Toto.
+
+The configurations (there are 2 options) are nearly the same as [ESM source and both CJS & ESM distribution](#esm-source-and-both-cjs-amp-esm-distribution), just exclude `packageJson.exports.import`.
+
+💡 Using `"type": "module"`[^2] paired with the `.cjs` file extension (for commonjs files) yields best results. For more information on why, see [Down the rabbit-hole](#down-the-rabbithole) and [Gotchas](#gotchas) below.
+
+**Working example**: [esm-with-cjs-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/esm/cjs-distro)
+
+### ESM source and both CJS & ESM distribution
+
+When source code is written in non-JavaScript (ex TypeScript), options can be limited due to needing to use file extension(s) specific to that language (ex `.ts`) and there may be no `.mjs` equivalent.
+
+Similar to [CJS source and both CJS & ESM distribution](#cjs-source-and-both-cjs-amp-esm-distribution), you have the same options.
+
+#### Publish only a CJS distribution with property exports
+
+The "Mojito" of packages: Tricky to make and needs good ingredients.
```

Comment:
Is this a cocktail reference?

Reply:
What the…I removed these…

Thanks for catching!